src/util.c

```diff
@@ -112,6 +113,29 @@ gchar *compute_strftime_suffix(const gchar *str, const gchar *strftime_suffix) {
     return g_strdup_printf("%s%s", str, outstr);
 }
 
+/**
+ * Compute absolute file path from directory path and file name
+ *
+ * @param directory absolute or relative directory path
+ * @param file_name file name (absolute or relative, if absolute directory is ignored)
+ * @return newly allocated string (free it with g_free) with absolute path to the file
+ */
+gchar *compute_file_path(const gchar *directory, const gchar *file_name) {
+    char cwd[PATH_MAX];
+    if ( file_name[0] == '/' ) {
+        return g_strdup_printf("%s", file_name);
+    }
+    else {
+        if ( directory[0] == '/' ) {
+            return g_strdup_printf("%s/%s", directory, file_name);
+        }
+        else {
+            getcwd(cwd, sizeof(cwd));
```

Comment:
use `g_get_current_dir()` 
https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-get-current-dir

instead of old `getcwd`?